libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2073,6 +2074,23 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 					errorMsg += " Use \"address(" + varName + ")." + memberName + "\" to access this address member.";
 					break;
 				}
+		}
+		if (exprType->category() == Type::Category::Address)
+		{
+			// Trigger error when using send or transfer with a non-payable fallback function.
+			auto addressType = dynamic_cast<AddressType const*>(exprType.get());
+			solAssert(!!addressType, "Should be address type.");
+
+			if (
+				(memberName == "send" || memberName == "transfer") &&
+				addressType->stateMutability() != StateMutability::Payable
+			)
+			{
+				errorMsg.pop_back();
+				errorMsg += " - Is the contract missing a payable fallback function?";
```

Comment:
I think it might even be better to replace the entire message here with
``"send" and "transfer" can only be called on "address payable".`` (or "on objects of type address payable" or something similar, but better ;-))

Not every address is a contract, so talking about contracts here might be misleading (think: ``address x = 0x1234...; x.send(...);`` - what contract are we talking about?). Searching the documentation for "address payable" should immediately clarify that a contract has a payable address iff it has a payable fallback function.